iceoryx_posh/test/moduletests/test_roudi_portmanager.cpp

```diff
@@ -585,6 +609,120 @@ TEST_F(PortManager_test, DoDiscoveryPublisherCanWaitAndSubscriberDiscardOldestLe
+    auto subscriberOptions = createTestSubOptions();
+
+    publisherOptions.historyCapacity = 0;
+    subscriberOptions.historyRequest = 1;
+    subscriberOptions.requiresPublisherHistorySupport = true;
+
+    auto publisher = createPublisher(publisherOptions);
+    auto subscriber = createSubscriber(subscriberOptions);
+
+    ASSERT_TRUE(publisher);
+    ASSERT_TRUE(subscriber);
+    EXPECT_FALSE(publisher.hasSubscribers());
+}
+
+TEST_F(PortManager_test, SubscriberNotRequiringHistorySupportDoesConnectToPublisherWithNoHistorySupport)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "080a94db-3a89-4d98-94a6-900015e608e2");
+
+    auto publisherOptions = createTestPubOptions();
+    auto subscriberOptions = createTestSubOptions();
+
+    publisherOptions.historyCapacity = 0;
+    subscriberOptions.historyRequest = 1;
+    subscriberOptions.requiresPublisherHistorySupport = false;
+
+    auto publisher = createPublisher(publisherOptions);
+    auto subscriber = createSubscriber(subscriberOptions);
+
+    ASSERT_TRUE(publisher);
+    ASSERT_TRUE(subscriber);
+    EXPECT_TRUE(publisher.hasSubscribers());
+}
+
+TEST_F(PortManager_test, SubscriberRequiringHistorySupportDoesConnectToPublisherWithSufficientHistorySupport)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "e2567667-4583-482b-9999-029f91c0cb71");
+
+    auto publisherOptions = createTestPubOptions();
+    auto subscriberOptions = createTestSubOptions();
+
+    publisherOptions.historyCapacity = 3;
+    subscriberOptions.historyRequest = 3;
+    subscriberOptions.requiresPublisherHistorySupport = true;
+
+    auto publisher = createPublisher(publisherOptions);
+    auto subscriber = createSubscriber(subscriberOptions);
+
+    ASSERT_TRUE(publisher);
+    ASSERT_TRUE(subscriber);
+    EXPECT_TRUE(publisher.hasSubscribers());
+}
+
+TEST_F(PortManager_test, SubscriberRequiringHistorySupportDoesNotConnectToPublisherWithInSufficientHistorySupport)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "20749a22-2771-4ec3-92f8-81bbdbd4aab6");
+
+    auto publisherOptions = createTestPubOptions();
+    auto subscriberOptions = createTestSubOptions();
+
+    publisherOptions.historyCapacity = 2;
+    subscriberOptions.historyRequest = 3;
+    subscriberOptions.requiresPublisherHistorySupport = true;
+
+    auto publisher = createPublisher(publisherOptions);
+    auto subscriber = createSubscriber(subscriberOptions);
+
+    ASSERT_TRUE(publisher);
+    ASSERT_TRUE(subscriber);
+    EXPECT_FALSE(publisher.hasSubscribers());
+}
+
+TEST_F(PortManager_test, SubscriberRequiringHistorySupportDoesNotConnectToPublisherWithNoHistorySupport)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "8f17d9cb-8f46-4742-99c3-b609a4d2319a");
+
+    auto publisherOptions = createTestPubOptions();
+    auto subscriberOptions = createTestSubOptions();
+
+    publisherOptions.historyCapacity = 0;
+    subscriberOptions.historyRequest = 3;
+    subscriberOptions.requiresPublisherHistorySupport = true;
```

Comment:
This is similar to the test at line 613 and following. Can this be removed?

Reply:
Yes, removed. It is the same equivalency class if nothing is broken. It might be broken for any arbitrary number though, in principle at least (but this cannot be feasibly checked).